db/helpers.js

```diff
@@ -0,0 +1,110 @@
+const mysql = require("mysql");
+const dotenv = require("dotenv");
+const moment = require("moment");
+const fs = require("fs");
+
+// initialize env variables
+dotenv.config();
+
+// connect to db
+const connection = mysql.createConnection({
+  host: process.env.DB_HOST,
+  user: process.env.DB_USER,
+  password: process.env.DB_PASSWORD,
+  database: "demozone_dev",
+});
+
+// store tokens in db
+const storeTokens = (installation) => {
+  let sql =
+    "INSERT INTO demo_app_tokens VALUES (?, ?, ?, ?, ?, ? ,?, ?, ?) ON DUPLICATE KEY UPDATE user_token= ?, bot_token = ?, installed_on = ?";
+  const inserts = [
+    "QUOTE-APPROVALS",
+    installation.user.id,
+    installation.team.id,
+    installation.team.name,
+    installation.enterprise.id,
+    installation.enterprise.name,
+    installation.user.token,
+    installation.bot.token,
+    moment().format("YYYY-MM-DD HH:mm:ss"),
+    installation.user.token,
+    installation.bot.token,
+    moment().format("YYYY-MM-DD HH:mm:ss"),
+  ];
```

Comment:
Don't we have `installation.user.token,
 installation.bot.token,
 moment().format("YYYY-MM-DD HH:mm:ss")` duplicated here?

Reply:
@clrmachado yes we do and there's a reason behind it

Each `?` in the `sql` statement is replaced in order by the items in the `inserts` array. The second reference in the array is required to substitute the `?`'s in the `ON DUPLICATE KEY` clause

Reply:
Thanks @georgefairbairn! Would never have understood that alone! 😃